daemon/api_snapshots.go

```diff
@@ -137,3 +142,49 @@ func changeSnapshots(c *Command, r *http.Request, user *auth.UserState) Response
 
 	return AsyncResponse(nil, &Meta{Change: chg.ID()})
 }
+
+type snapshotExportResponse struct {
+	SetID      uint64
+	ExportSize uint64
+}
+
+func (s snapshotExportResponse) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Length", strconv.FormatUint(s.ExportSize, 10))
+
+	snapshotExport(context.TODO(), uint64(s.SetID), w)
+}
+
+type countingOnlyWriter struct {
+	total uint64
+}
+
+func (w *countingOnlyWriter) Write(p []byte) (n int, err error) {
+	w.total += uint64(len(p))
+	return len(p), nil
+}
+
+func getSnapshotExport(c *Command, r *http.Request, user *auth.UserState) Response {
+	vars := muxVars(r)
+	sid := vars["id"]
+	setID, err := strconv.ParseUint(sid, 10, 64)
+	if err != nil {
+		return BadRequest("'id' must be a positive base 10 number; got %q", sid)
+	}
+
+	// Export into /dev/null once to get the size of the tar so that
```

Comment:
nitpick, this doesn't actually export literally into /dev/null, it just never uses the data so the go runtime garbage collector deletes it. I was rather confused reading this code where /dev/null entered the picture until I realized this was just a way of saying that we aren't using the data on this first go around.

Reply:
I updated the comment to make this clearer. Sorry for the confusion.